### PR TITLE
Add missing Docker Containers and Fix SONAR build

### PR DESCRIPTION
**Pull request by jimzucker**
Fixed the Sonar build and updated docker containers for all modules.

* #337 - Add Docker Containers for stash/svn/udeploy/versionone & cleanup docker file for jenkins test instance (5 days ago) Jim Zucker

* #363 - updated the jacoco plugin to the current version (8 days ago) Jim Zucker

* #364 Upgrade maven-compiler-plugin to 3.5.1 rev

*  #365 Jacoco not running for all subproject -  Fixed jacoco/Sonar runs.  There is one outstanding issue and sonar is skipped for the git and stash collectors(ref #370)  

*  #351 - PMD Warning - Overridable method called during object construction - had to fix them in this branch as build was failing after compiler upgrade (9 days ago) Jim Zucker

It is safe to commit but I need help testing these containers: 
jira
version-one
subversion
udeploy

if somebody can give me details what to setup I am happy to test and I can create esc environments for them like I did for Jenkins & Sonar.
screenshot 2016-03-04 12 40 32

screenshot 2016-03-04 12 40 40

### CODE REVIEW

File: stash-scm-collector/pom.xml [hunk 1]
```diff
@@ -101,6 +138,12 @@
 			<groupId>joda-time</groupId>
 			<artifactId>joda-time</artifactId>
 		</dependency>
-	</dependencies>
+		<dependency>
+		    <groupId>org.joda</groupId>
+		    <artifactId>joda-convert</artifactId>
+		    <version>1.8.1</version>
```
Comment (jdamick) [inline stash-scm-collector/pom.xml]: why r u locking the version here?

File: github-scm-collector/pom.xml [hunk 2]
```diff
@@ -115,6 +134,12 @@
 			<groupId>joda-time</groupId>
 			<artifactId>joda-time</artifactId>
 		</dependency>
+		<dependency>
+		    <groupId>org.joda</groupId>
+		    <artifactId>joda-convert</artifactId>
+		    <version>1.8.1</version>
```
Comment (jdamick) [inline github-scm-collector/pom.xml]: let's create a property for it..

**Comment by jdamick** [comment]
cool thanks
i see we need a better sol'n for the dockerfiles, they are all copied & pasted from the original couple ..

File: stash-scm-collector/pom.xml (hunk 1 above)
Comment (jimzucker) [inline stash-scm-collector/pom.xml]: fixed

File: github-scm-collector/pom.xml (hunk 2 above)
Comment (jimzucker) [inline github-scm-collector/pom.xml]: done

**Comment by jimzucker** [comment]
@jdamick I updated per the comments.  Re the docker files, I dont think there is anything to do there, its how they work...  if we had redundant layers we could create one and then build from it but that is not the case here..

**Comment by jimzucker** [comment]
@jdamick @tabladrum  - I see this has conflicts now, do you need me to rebase it ?

**Comment by tabladrum** [comment]
That will be great.

**Comment by jimzucker** [comment]
@tabladrum done, I created a new branch & Pull request, see #433

**Pull request closed without merging**